brickflow/engine/task.py

```diff
@@ -822,15 +822,24 @@ def parents(self) -> List[str]:
     @property
     def depends_on_names(self) -> Iterator[Dict[str, Optional[str]]]:
         for i in self.depends_on:
-            if self.if_else_outcome:
-                outcome = list(self.if_else_outcome.values())[0]
+            task_name = i.__name__ if callable(i) and hasattr(i, "__name__") else str(i)
```

Comment:
you end up testing `if callable(i) and hasattri(i, "__name__")` twice per loop. 
You could store in a `task_is_callable`  variable and then use it like.

~~~python
task_is_callable = True if callable(i) and hasattri(i, "__name__") else False
task_name = i.__name__ if task_is_callable else str(i)
...
if task_is_callable:
  yield {task_name: outcome}
else:
  yield {task_name: outcome}
~~~

However, given you only really need to test once when you set `task_name` on line 825,
the final block could just be condensed to `yield { task_name: outcome }`

Reply:
Fixed, thanks!